src/tbb/arena.h

```diff
@@ -356,8 +360,11 @@ class arena: public padded<arena_base>
     //! If necessary, raise a flag that there is new job in arena.
     template<arena::new_work_type work_type> void advertise_new_work();
 
+    //! Get gets the index of a non-empty task pool
+    long long get_index_for_steal(FastRandom& frnd, std::size_t slot_num_limit);
```

Comment:
Why public? Maybe move this function into `steal_task` as lambda?

Reply:
1) I would suggest not expand steal_task size(amount of lines) and create new method.
2) Because my_limit is atomic, so we get snapshot in steal_task and pass it to help method.

Reply:
@pavelkumbrasev why we cant get snapshot in get_index_for_steal ? we dont use it in steal task